overlord/devicestate/handlers_install.go

```diff
@@ -126,6 +126,56 @@ func writeLogs(rootdir string) error {
 	if err := gz.Flush(); err != nil {
 		return fmt.Errorf("cannot flush compressed log output: %v", err)
 	}
+
+	return nil
+}
+
+func writeTimings(rootdir string, chgIDs []string) error {
+	logPath := filepath.Join(rootdir, "var/log/install-timings.txt.gz")
+	if err := os.MkdirAll(filepath.Dir(logPath), 0755); err != nil {
+		return err
+	}
+
+	f, err := os.Create(logPath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	gz := gzip.NewWriter(f)
+	defer gz.Close()
+
+	// XXX: ugly, ugly, but using the internal timings requires
+	//      some refactor as a lot of the required bits are not
+	//      exported right now
+	// first all changes
+	cmd := exec.Command("snap", "changes")
```

Comment:
For completeness, would it be useful to add a line with the command to the output file first? eg.
```
fmt.Fprintf(gz, "---- Output of: snap changes\n")
...
fmt.Fprintf(gz, "---- Output of snap debug timings --ensure=seed")
...
```